esp/scm/ws_sasha.ecm

```diff
@@ -0,0 +1,108 @@
+{
+    ECL("ECL"),
+    DFU("DFU"),
+};
+
+ESPrequest [nil_remove] GetVersionRequest
+{
+};
+
+ESPresponse [exceptions_inline, nil_remove] ResultResponse
+{
+    string Result;
+};
+
+ESPrequest [nil_remove] ListWURequest
+{
+    ESPenum WUTypes WUType;
+    string Wuid;
+    string Cluster;
+    string Owner;
+    string JobName;
+    string State;
+    string AfterDate; //YYYYMMDD
+    string BeforeDate; //YYYYMMDD
+
+    bool Archived(true);
+    bool Online(true);
+
+    bool IncludeDT(false);
+    string OutputFields;
+    int64 PageSize(0);
+    int64 PageStartFrom(0);
+};
+
+ESPrequest [nil_remove] ArchiveWURequest
+{
+    string Wuid;
+    ESPenum WUTypes WUType;
+};
+
+ESPrequest [nil_remove] RestoreWURequest
+{
+    string Wuid;
+    ESPenum WUTypes WUType;
+};
+
+ESPrequest [nil_remove] XREFRequest
+{
+    string Cluster;
+};
+
+ESPservice [auth_feature("DEFERRED"), //This declares that the method logic handles feature level authorization
+    version("1.00"), default_client_version("1.00"), exceptions_inline("./smc_xslt/exceptions.xslt")] WSSasha
+{
+    ESPmethod [auth_feature("SashaAccess:Access")] GetVersion(GetVersionRequest, ResultResponse);
+    ESPmethod [auth_feature("SashaAccess:READ")] ListWU(ListWURequest, ResultResponse); //list ECL WUs or DFU WUs in Archive server.
+    //In a separate PR, new method DownloadWU will be added to output to file.
+
+    //ESPmethod [auth_feature("SashaAccess:READ")] Get(GetRequest, ResultResponse);
+        // By reading the code, looks like that it lists ECL WUs or DFU WUs in Archive server in XML format.
+        // When I tested using the existing Sasha CLI, it does not work. ?
+        // If works, should it be merged into the ListWU as an option (Format:XML?)?
+
+    //ESPmethod [auth_feature("SashaAccess:READ")] ListDT(ListDTRequest, ResultResponse);
+        // By reading the code, looks like: it lists ECL WUs or DFU WUs with date time (when?) in Archive server.
+        // Should it be merged into the ListWU as an option (IncludeDateTime)?
+
+    ESPmethod [auth_feature("SashaAccess:FULL")] ArchiveWU(ArchiveWURequest, ResultResponse); //archive ECL WUs or DFU WUs
+    ESPmethod [auth_feature("SashaAccess:FULL")] RestoreWU(RestoreWURequest, ResultResponse); //restore ECL WUs or DFU WUs
+
+    //ESPmethod [auth_feature("SashaAccess:FULL")] BackupWU(BackupWURequest, ResultResponse); //backup one ECL WU
+    // Only backup an ECL WU (not DFU WU). How does it work? How to test? How about implementing in separate PR?
+
+    ESPmethod [auth_feature("SashaAccess:FULL")] XREF(XREFRequest, ResultResponse); //generate XREF for clusters
+
+    //ESPmethod [auth_feature("SashaAccess:FULL")] Stop(StopRequest, ResultResponse);
+    // Stop what? How does it work? How to test? How about implementing in separate PR?
+
+    //Do we need the following methods? They are not in the Sasha Usage, but, in Sasha code. If yes, how about implementing in separate PRs?
+        //SCA_WORKUNIT_SERVICES_GET ?
```

Comment:
I don't think needed by CLI or service, but still needed, i.e. components (like esp) that interact with Sasha directly, should still continue to allow this command to be created and sent to Sasha directly.

Reply:
What does the SCA_WORKUNIT_SERVICES_GET (as well as coalesce_suspend and coalesce_resume) do?

Reply:
>SCA_WORKUNIT_SERVICES_GET

I can't recall exactly, but I don't think this new service needs to support it for now, it will not be something wanted by the CLI or UI, but instead used by other esp services that talk directly to sasha (afaics only used by wsWorkunitList atm).

Reply:
> coalesce_suspend and coalesce_resume) 

One of sasha's function is to coalesce the dali store (aggregate the current store + in flight deltas into a new store file).
It is resource intensive, and I think the idea was to pause it, so other resource intensive sasha services like XREF, could have more capacity.
In cloud, I don't think coalesce_suspend/coalesce_resume make much sense, since each sasha service is in it's own dedicated container.